src/Database/StatementInterface.php

```diff
@@ -16,13 +16,15 @@
  */
 namespace Cake\Database;
 
+use Traversable;
+
 /**
  * Represents a database statement. Concrete implementations
  * can either use PDOStatement or a native driver
  *
  * @property-read string $queryString
  */
-interface StatementInterface
+interface StatementInterface extends Traversable
```

Comment:
This will get further merged into StatementDecorator which implements IteratorAggregate so this should be fine.